api/src/main/java/org/openmrs/module/fhir/api/strategies/condition/ConditionStrategy.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+
+package org.openmrs.module.fhir.api.strategies.condition;
+
+import org.apache.commons.lang.ArrayUtils;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.openmrs.Obs;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.fhir.api.util.ContextUtil;
+import org.openmrs.module.fhir.api.util.FHIRConditionUtil;
+import org.openmrs.module.fhir.api.util.FHIRUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component("DefaultConditionStrategy")
+public class ConditionStrategy implements GenericConditionStrategy{
+
+	@Override
+	public Condition getConditionById(String uuid) {
+		return ContextUtil.getConditionHelper().getCondition(uuid);
+	}
+
+	@Override
+	public Condition getConditionByObsId(String uuid) {
+		int[] conceptsAsConditions = FHIRUtils.getConceptIdsOfConditions();
+		Obs obs = Context.getObsService().getObsByUuid(uuid);
+		if (obs == null || obs.isVoided() || conceptsAsConditions == null || !ArrayUtils.contains
+				(conceptsAsConditions, obs.getConcept().getId())) {
+			return null;
+		}
+
+		return FHIRConditionUtil.generateFHIRConditionFromOpenMRSObs(obs);
+	}
+
+	@Override
+	public List<Condition> searchConditionById(String uuid) {
+		return null;
```

Comment:
Are you intentionally returning null in these two methods?

Reply:
yeah, I will work on them in the next ticket. I am only handling create method and fixing getbyId
https://issues.openmrs.org/browse/FM-65

Reply:
Shouldn't we throw NotImplementedException instead of returning null?

Reply:
I mean throwing NotImplementedException instead of returning null